solarforecastarbiter/metrics/calculator.py

```diff
@@ -3,98 +3,161 @@
 
 Right now placeholder so we can delete report.metrics.py.
 Needs to cleaned up and expanded.
+
+Todo
+----
+* Support probabilistic metrics and forecasts with new functions
+* Support event metrics and forecasts with new functions
 """
 from collections import defaultdict
 
-import numpy as np
 import pandas as pd
 
+from solarforecastarbiter import datamodel
 from solarforecastarbiter.metrics import deterministic
 
 
-def calculate_metrics_for_processed_pairs(processed_fxobs):
+AVAILABLE_CATEGORIES = [
+    'total', 'year', 'month', 'day', 'hour', 'date', 'weekday'
+]
+
+
+def calculate_metrics(processed_pairs, categories, metrics,
+                      ref_pair=None, normalizer=1.0):
     """
     Loop through the forecast-observation pairs and calculate metrics.
 
     Parameters
     ----------
-    processed_fxobs : list
+    processed_pairs :
         List of solarforecastarbiter.datamodel.ProcessedForecastObservation.
+    categories : list of str
+        List of categories to compute metrics over.
+    metrics : list of str
+        List of metrics to be computed.
+    ref_fx_obs :
+        solarforecastarbiter.datamodel.ProcessedForecastObservation`
+        Reference forecast to be used when calculating skill metrics. Default
+        is None and no skill metrics will be calculated.
+    normalizer : float
+        Normalized factor (should be in the same units as data). Default is
+        1.0 and only needed for normalized metrics.
 
     Returns
     -------
-    dict
-        Multi-level dictionary of metrics by category.
+    list
+        List of dict with the metric results and a key with original forecast
+        name for identification.
+
+    Todo
+    ----
+    * validate categories are supported
+    * validate metrics are supported
+    * Support probabilistic metrics and forecasts
+    * Support event metrics and forecasts
     """
     calc_metrics = []
-    for fxobs_ in processed_fxobs:
-        metrics_ = calculate_metrics(fxobs_.original,
-                                     fxobs_.forecast_values,
-                                     fxobs_.observation_values)
-        calc_metrics.append(metrics_)
-    return calc_metrics
 
+    for proc_fxobs in processed_pairs:
+
+        # Deterministic
+        if isinstance(proc_fxobs, datamodel.ProcessedForecastObservation):
```

Comment:
I will certainly add that check for now. Also how will we distinguish between Event-type forecasts and observations from deterministic?  Right now I don't see anything in datamodels that distinguishes it.

Reply:
We've not put much thought into implementing support for event forecasts and observations. I think it might become more clear once probabilistic forecasts are better supported, so let's ignore them for this PR.

Reply:
Ok. Thanks